addons/purchase_requisition/models/purchase_requisition.py

```diff
@@ -98,6 +77,12 @@ def _onchange_vendor(self):
             }
             return {'warning': warning}
 
+    @api.onchange('requisition_type')
+    def _onchange_requisition_type(self):
+        if self.line_ids:
+            for line in self.line_ids:
+                line._onchange_product_id()
```

Comment:
I tried this suggestion but it's not working. This may be due to `def _onchange_requisition_type()` is in `purchase.requisition`
and `_onchange_product_id()` is in `purchase.requisition.line`
 onchange, depends are not working on _onchange_product_id()